### PR TITLE
Use AsyncAws to get more light weight dependencies

### PR DESCRIPTION
**Pull request by Nyholm**
Since Lambda has a limit on how big applications can be, I thought it would be a good idea to reduce the dependencies. 

This introduces [AsyncAws](https://async-aws.com/) instead of the official AWS SDK. 

The no-dev dependencies in ./vendor: 
- Before: 44Mb
- After: 26Mb

------

The `AsyncAwsSqsQueue` and `AsyncAwsSqsJob` could be moved to a different package if you prefer. 

-------

This PR is not complete. I will fix the tests and make sure to release stable version of Lambda and SSM client. I would like some thumbs up or down if you would like this feature or not.

### CODE REVIEW

File: src/Http/Controllers/SignedStorageUrlController.php [hunk 1]
```diff
@@ -112,26 +110,23 @@ protected function ensureEnvironmentVariablesAreAvailable(Request $request)
...
     protected function storageClient()
     {
         $config = [
             'region' => $_ENV['AWS_DEFAULT_REGION'],
-            'version' => 'latest',
-            'signature_version' => 'v4',
         ];
 
         if (! isset($_ENV['AWS_LAMBDA_FUNCTION_VERSION'])) {
             $config['credentials'] = array_filter([
```
Comment (jderusse) [inline src/Http/Controllers/SignedStorageUrlController.php:122]: `$config = [`

File: src/Queue/AsyncAwsSqsQueue.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+     * Get the size of the queue.
+     *
+     * @param  string|null  $queue
+     * @return int
+     */
+    public function size($queue = null)
+    {
+        $response = $this->sqs->getQueueAttributes([
+            'QueueUrl' => $this->getQueue($queue),
+            'AttributeNames' => ['ApproximateNumberOfMessages'],
```
Comment (jderusse) [inline src/Queue/AsyncAwsSqsQueue.php:59]: Could use the constant instead `QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES`

File: src/Queue/AsyncAwsSqsQueue.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     * Pop the next job off of the queue.
+     *
+     * @param  string|null  $queue
+     * @return \Illuminate\Contracts\Queue\Job|null
+     */
+    public function pop($queue = null)
+    {
+        $response = $this->sqs->receiveMessage([
+            'QueueUrl' => $queue = $this->getQueue($queue),
+            'AttributeNames' => ['ApproximateReceiveCount'],
```
Comment (jderusse) [inline src/Queue/AsyncAwsSqsQueue.php:123]: Shouldn't we leverage `WaitTimeSeconds` and `MaxNumberOfMessages`?

File: src/Queue/AsyncAwsSqsQueue.php (hunk 2 above)
Comment (Nyholm) [reply src/Queue/AsyncAwsSqsQueue.php:59]: Sure. Thank you

File: src/Http/Controllers/SignedStorageUrlController.php (hunk 1 above)
Comment (Nyholm) [reply src/Http/Controllers/SignedStorageUrlController.php:122]: Thanks!

**Comment by viezel** [comment]
Yes please 👍🏻

**Comment by taylorotwell** [comment]
Just being honest I don't see myself going this direction. It's a bit risky for me to adopt this as we're already managing 4000+ Lambda projects on Vapor and for that amount of customers I can't risk breaking their applications.

Some fears I have are produced by this portion of the website:

image

First, you guys aren't dumb and lazy I hope 😆 and the "no frequent updates" and "not feature complete" give me pause as well. 

I think AsyncAws is a cool project idea though and I may very well explore it again in the future as it continues to mature.

Thanks!

**Comment by Nyholm** [comment]
Thank you for the quick reply. I very much understand your concern and I think they are valid. 

About "no frequent updates": We have a bot that keeps us up to date with API changes. We should maybe rename it to "only relevant updates". 

About "not feature complete": We are happy to generate more features, but only features that somebody ask for. 

I hope you don't mind if I make a similar PR next year. Until then we will focus on getting more attraction and create better Laravel integrations. We just started With `Filesystem`, `Queue` and `Mail`: https://github.com/async-aws?q=illuminate&type=&language=

Keep up the good work!

**Pull request closed without merging**